scaleway/resource_container_test.go

```diff
@@ -0,0 +1,392 @@
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource scaleway_container_namespace main {
+					}
+
+					resource scaleway_container main {
+						namespace_id = scaleway_container_namespace.main.id
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayContainerExists(tt, "scaleway_container.main"),
+					testCheckResourceAttrUUID("scaleway_container_namespace.main", "id"),
+					testCheckResourceAttrUUID("scaleway_container.main", "id"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "name"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "registry_image"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "domain_name"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "max_concurrency"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "domain_name"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "protocol"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "cpu_limit"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "timeout"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "memory_limit"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "max_scale"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "min_scale"),
+					resource.TestCheckResourceAttrSet("scaleway_container.main", "privacy"),
+				),
+			},
+			{
+				Config: `
+					resource scaleway_container_namespace main {
+					}
+
+					resource scaleway_container main {
+						name = "my-container-tf"
+						namespace_id = scaleway_container_namespace.main.id
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayContainerExists(tt, "scaleway_container.main"),
+					testCheckResourceAttrUUID("scaleway_container.main", "id"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "name", "my-container-tf"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "port", "8080"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "cpu_limit", "70"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "memory_limit", "128"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "min_scale", "0"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "max_scale", "20"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "timeout", "300"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "max_concurrency", "50"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "redeploy", "false"),
+					resource.TestCheckResourceAttr("scaleway_container.main", "privacy", container.ContainerPrivacyPublic.String()),
+					resource.TestCheckResourceAttr("scaleway_container.main", "protocol", container.ContainerProtocolHTTP1.String()),
+				),
+			},
+			{
+				Config: `
+					resource scaleway_container_namespace main {
```

Comment:
What is the difference between this step and the step before?

Reply:
This was added to force the `registry_namespace` created as default. I'm gonna remove this and the namespaces are going to be treated by the sweeper process.